File: 백준/Gold/1911. 흙길 보수하기/흙길 보수하기.py
N, L = map(int, input().split())

p = []

# 튜플 형식으로 list에 입력 받기
for i in range(N):
    a, b = map(int, input().split())
    p.append((a, b))

p.sort()  # 오름차순 정렬

now = 0
cnt = 0
for i in range(N):
    s = max(now, p[i][0])
    e = p[i][1]

    if s >= e:
        continue

    gap = e - s

    if gap % L == 0:
        need = int(gap / L)
    else:
        need = int(gap / L) + 1

    cnt += need
    now = s + need * L


print(cnt)

File: 백준/Gold/1911. 흙길 보수하기/흙길 보수하기.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    int N, L;
    cin >> N >> L;

    vector<pair<int, int>> p(N);
    for (int i = 0; i < N; i++)
    {
        cin >> p[i].first >> p[i].second;
    }

    sort(p.begin(), p.end());

    int now = 0, cnt = 0;
    for (int i = 0; i < N; i++)
    {
        int s = max(now, p[i].first);
        int e = p[i].second;

        if (s >= e)
        {
            continue;
        }

        int len = e - s;
        int need = (len % L == 0 ? len / L : len / L + 1);

        cnt += need;
        now = s + need * L;
    }
    cout << cnt;
}